packages/api/src/controllers/region.js

```diff
@@ -0,0 +1,63 @@
+import Router from 'express/lib/router'
+import { authMiddleware } from '../middleware'
+import { db } from '../store'
+
+const app = Router()
+
+app.get('/', authMiddleware({ admin: true }), async (req, res, next) => {
+  const [regions, cursor] = await db.region.find()
+  if (req.query.grouped) {
+    return res.json(regions)
+  }
+
+  const flatOrchList = []
+  regions.forEach((region) => {
+    region.orchestrators.forEach((orch) => {
+      orch.region = region.region
+      flatOrchList.push(orch)
+    })
+  })
+
+  return res.json(flatOrchList)
+})
+
+app.get('/:region', authMiddleware({ admin: true }), async (req, res, next) => {
+  const region = await db.region.get(req.params.region)
+  return res.json(region)
+})
+
+app.put('/:region', authMiddleware({ admin: true }), async (req, res, next) => {
+  if (!req.body || !req.body.region) {
+    res.status(406)
+    return res.json({ error: 'region field is required' })
+  }
+
+  if (!req.body.orchestrators || !req.body.orchestrators.length) {
+    console.log(`WARNING req.body.orchestrators is undefined`)
+  }
+
+  let region = {
+    region: req.body.region,
+    orchestrators: req.body.orchestrators || [],
+  }
+
+  const resp = await db.region.create({
```

Comment:
So we don't want to create a new region every time. The main use case here: we're going to run a daemon in each region that runs `PUT /regions/:regionName` every minute or so. So this will need to be an "upsert" operation — first try and get the region. If it already exists, do a `db.region.update({})` with the new list of Os. If it doesn't, `db.region.create({})` just like this.